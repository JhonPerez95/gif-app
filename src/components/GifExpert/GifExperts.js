import React, { useState } from 'react';
import CategoryAdd from '../CategoryAdd/CategoryAdd';
import GifGrid from '../GifGrid/GifGrid';

const GifExperts = ({ defaultCategory = ['naruto'] }) => {
  const [categories, setCategories] = useState(defaultCategory);

  return (
    <>
      <h1>GifExperts</h1>
      <CategoryAdd setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((item) => (
          <GifGrid key={item} category={item} />
        ))}
      </ol>
    </>
  );
};

export default GifExperts;
