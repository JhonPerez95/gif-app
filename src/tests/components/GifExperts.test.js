import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import GifExperts from '../../components/GifExpert/GifExperts';

describe('Testing al componente GifExperts: ', () => {
  test('Renderizado correctamente del componente <GifExpert/> ', () => {
    const wrapper = shallow(<GifExperts />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar GifGrid  que estan por defecto', () => {
    const category = ['Naruto', 'Bleach', 'Black Clover'];
    const wrapper = shallow(<GifExperts defaultCategory={category} />);

    expect(wrapper.find('GifGrid').length).toBe(category.length);
    expect(wrapper).toMatchSnapshot();
  });
});
