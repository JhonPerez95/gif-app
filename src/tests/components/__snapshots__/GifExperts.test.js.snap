// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing al componente GifExperts:  Debe mostrar GifGrid  que estan por defecto 1`] = `
<Fragment>
  <h1>
    GifExperts
  </h1>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Bleach"
      key="Bleach"
    />
    <GifGrid
      category="Black Clover"
      key="Black Clover"
    />
  </ol>
</Fragment>
`;

exports[`Testing al componente GifExperts:  Renderizado correctamente del componente <GifExpert/>  1`] = `
<Fragment>
  <h1>
    GifExperts
  </h1>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="naruto"
      key="naruto"
    />
  </ol>
</Fragment>
`;
